Sources/HTTP/SimpleSocket/SimpleServerSocket.swift

```diff
@@ -0,0 +1,152 @@
+// This source file is part of the Swift.org Server APIs open source project
+//
+// Copyright (c) 2017 Swift Server API project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+//
+
+import Foundation
+import Dispatch
+
+internal class SimpleServerSocket {
+    internal var socketfd: Int32 = -1
+    
+    internal var listeningPort: Int32 = -1
+    
+    internal private(set) var isListening = false
+    
+    internal private(set) var isConnected = false
+
+    internal func socketRead(into readBuffer: inout UnsafeMutablePointer<Int8>, maxLength:Int) throws -> Int {
+        //let readBuffer: UnsafeMutablePointer<CChar> = UnsafeMutablePointer<CChar>.allocate(capacity: maxLength)
+        readBuffer.initialize(to: 0x0)
+        let read = recv(self.socketfd, readBuffer, maxLength, Int32(0))
```

Comment:
Just `return recv(...)`

Reply:
I like putting a breakpoint on the return while troubleshooting and being able to see/print the value.  Harder to do if there's no local var (and the optimizer should make them perform equivalently).

Reply:
You can set a breakpoint then step in/out a few times until `recv()` returns, but OK.